iothub/service/src/DigitalTwin/readme.md

```diff
@@ -1,20 +1,91 @@
-#### Autorest config file for generating Digital Twin Service Client models and protocol layer
+  > The `DigitalTwinClient` is available only on .NET framework 4.7.2 and .NET Standard 2.0+.
 
-#### Protocol layer generation:
-- Run the powershell script `generateCode.ps`. It will pick up the config below and output the results into this folder. It will also make a few automated changes to the generated protocol layer, that are required for this client library.
+### Protocol layer generation:
+- Run the powershell script [generateCode.ps](./generateCode.ps1). It will pick up the autorest config and output the results into this folder. It will also make a few automated changes to the generated protocol layer, that are required for this client library.
 
-*NOTE: The `DigitalTwinClient` is available only on .NET framework 4.7.2 and .NET Standard 2.0+.
+### Examples
 
+You can familiarize yourself with different APIs using [samples for Digital Twins](https://github.com/Azure-Samples/azure-iot-samples-csharp/tree/feature/digitaltwin/iot-hub/Samples/service/DigitalTwinClientSamples).
 
-> see https://aka.ms/autorest
+## Source code folder structure
 
-``` yaml 
-input-file: DigitalTwin.json
+### /Authentication
+
+The code for generating shared access signature tokens, used for authentication Http requests against Azure IoT Hub service.
+
+### /Generated
+
+The code generated by autorest using the swagger file defined under [DigitalTwin.json](./DigitalTwin.json).
+
+### /src/Customized
+
+The customzied code written to override the following behavior of auto-generated code:
+
+- Rename some of the generated types, eg. [DigitalTwinInvokeCommandHeaders](./Customized/DigitalTwinInvokeCommandHeaders.cs)
```

Comment:
Should we really give all these details here? This is implementation details for what we generate. Customers need to adhere to this if they want to generate their own SDK. The things below does not tell why we do it so it might look confusing. Thoughts?

Reply:
I followed the format that we have in track 2 repo.
I don't mind putting this info in, as it find of explains why we have a "customization" on top. If they look at the "generateCode" script, they find the same info = what is being done, but not why.